ui/src/app/workflows/components/workflow-timeline/workflow-timeline.tsx

```diff
@@ -18,128 +18,124 @@ interface WorkflowTimelineProps {
     nodeClicked?: (node: models.NodeStatus) => any;
 }
 
-interface WorkflowTimelineState {
-    parentWidth: number;
-    now: moment.Moment;
-}
+export function WorkflowTimeline(props: WorkflowTimelineProps) {
+    let resizeSubscription: Subscription;
+    let refreshSubscription: Subscription;
```

Comment:
```suggestion
    const resizeSubscription = useRef<Subscription>(null);
    const refreshSubscription = useRef<Subscription>(null);
```

these actually should be refs as well so that they are persisted between renders.
Right now I think it only still works by coincidence, since all the functions have it in their scope, but the best practice (and least error prone) here would be to use refs

Reply:
Could you also move these to be together with `containerRef` as they were before? so that all refs are together